### DISCUSSION TITLE
2. 데이터베이스 - SQL 기초

### DISCUSSION THREAD

**Discussion by zini9188** [Shortudy-10th/tech-for-developer#151] [질의 응답]
- 데이터 조작어-검색
- 데이터 정의어
- 데이터 조작어-삽입, 수정, 삭제
 - 내장 함수
- 그 외 등등

**Comment by Jihn0118** [comment]
## Q. SELECT 쿼리의 수행 순서를 알려주세요.

**FROM, JOIN > WHERE > GROUP BY > HAVING > SELECT > DISTINCT > ORDER BY > LIMIT 순서** 입니다.

테이블을 확인하고 
JOIN 조건 확인하고 JOIN 실행되어 데이터가 하나의 가상 테이블로 모아지게 되고
지정한 테이블에서 조건에 맞는 데이터를 필터링하고
선택한 컬럼을 기준으로 조건에 맞게 그룹핑 하고
여러 조건들을 처리한 후 남은 데이터에서 어떤 열을 출력해줄지 선택하고
정렬해주고
결과 중 몇 개의 행을 보여줄 지 선택합니다.

**Reply by zini9188** [reply]
where -> group by -> having

**Comment by zini9188** [comment]
### Q. DDL에 대해 설명해주세요.
객체의  생성, 변경, 삭제 명령어를 뜻하며, create, alter, drop, rename 등이 있습니다. 데이터 정의어는 3가지 유형으로 create, alter, drop이 있습니다.

**Reply by watty-exclamationMark** [reply]
## Q. DML은 무엇인가요?
데이터 조작을 위한 언어로,  select, insert, update, delete가 존재합니다.

**Reply by seongeun42** [reply]
## Q. DCL / TCL은 무엇인가요?

**DCL**(Data Control Language)은 데이터를 제어하는 언어로, 데이터의 보안, 무결성, 회복, 병행 수행제어 등을 정의하는 데 사용합니다. 작업 결과를 반영하는 **commit**, 작업을 취소하고 복구하는 **rollback**, 사용자에게 권한을 부여하는 **grant**, 권한을 취소하는 **revoke**가 있습니다. 이때 트랜잭션을 제어하는 commit과 rollback의 경우 따로 **TCL**(Transaction Control Language)라고 하기도 합니다.

**Comment by watty-exclamationMark** [comment]
## Q. JOIN에 대해 설명해주세요.
두 개 이상의 Table이나 Database를 연결하여 데이터를 검색할 때 사용하는 방법입니다.

**Comment by seongeun42** [comment]
## Q. Drop, Delete, Truncate의 차이에 대해 말해주세요.

**Drop**은 테이블 자체를 삭제하는 명령어로, 자동 Commit 되기 때문에 복구할 수 없습니다. **Delete**는 테이블 내에 저장된 데이터를 삭제하는 명령어입니다. **Truncate**는 테이블에 저장된 모든 데이터를 삭제하는 명령어로, Delete와 달리 자동 Commit 되기 때문에 복구할 수 없습니다.
